GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1780,19 +1780,31 @@ private FSResult RemoveFolderPlaceholderIfEmpty(IPlaceholderData placeholder)
 
                 default:
                     {
-                        string gitPath;
-                        this.AddFileToUpdateDeletePlaceholderFailureReport(deleteOperation, placeholder, out gitPath);
                         this.ScheduleBackgroundTaskForFailedUpdateDeletePlaceholder(placeholder, deleteOperation);
                         this.AddParentFoldersToListToKeep(parentKey, folderPlaceholdersToKeep);
 
                         metadata = CreateEventMetadata();
                         metadata.Add("deleteOperation", deleteOperation);
                         metadata.Add("virtualPath", placeholder.Path);
-                        metadata.Add("gitPath", gitPath);
                         metadata.Add("result.Result", result.ToString());
                         metadata.Add("result.RawResult", result.RawResult);
                         metadata.Add("failureReason", failureReason.ToString());
-                        this.context.Tracer.RelatedWarning(metadata, "UpdateOrDeletePlaceholder: did not succeed");
+
+                        bool logOnly = result.Result == FSResult.GenericProjFSError && failureReason == UpdateFailureReason.DirtyData;

```

Comment:
We can add
```
case FSResult.GenericProjFSError when failureReason == UpdateFailureReason.DirtyData:
case FSResult.VirtualizationInvalidOperation:
```
so it will fall into the same code as `VirtualizationInvalidOperation` and the `default` will not have to change.